### PR TITLE
AMDGPU/NewPM: Port SIAnnotateControlFlow to new pass manager

### PR DESCRIPTION
**Pull request by arsenm**
Does not yet add it to the pass pipeline. Somehow it causes
2 tests to assert in SelectionDAG, in functions without any
control flow.

### CODE REVIEW

**Comment by arsenm** [comment]
* **#102654** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/102654?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#102653** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/102653?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @arsenm and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: llvm/lib/Target/AMDGPU/SIAnnotateControlFlow.cpp [hunk 1]
```diff
@@ -349,15 +331,9 @@ bool SIAnnotateControlFlow::closeControlFlow(BasicBlock *BB) {
...
-  DT = &getAnalysis<DominatorTreeWrapperPass>().getDomTree();
-  LI = &getAnalysis<LoopInfoWrapperPass>().getLoopInfo();
-  UA = &getAnalysis<UniformityInfoWrapperPass>().getUniformityInfo();
-  TargetPassConfig &TPC = getAnalysis<TargetPassConfig>();
-  const TargetMachine &TM = TPC.getTM<TargetMachine>();
-
+bool SIAnnotateControlFlow::run(Function &F) {
   bool Changed = false;
-  initialize(*F.getParent(), TM.getSubtarget<GCNSubtarget>(F));
+
```
Comment (gandhi56) [inline llvm/lib/Target/AMDGPU/SIAnnotateControlFlow.cpp:336]: `initialize(..)` does not need to be called anymore?

Comment (arsenm) [reply llvm/lib/Target/AMDGPU/SIAnnotateControlFlow.cpp:336]: It's now done in the constructor

**Review by gandhi56 — APPROVED** [review]

**Comment by arsenm** [comment]
### Merge activity

* **Aug 9, 10:57 PM EDT**: @arsenm started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/llvm/llvm-project/102653).

**Pull request merged**